lua/chatcommands/duel.lua

```diff
@@ -0,0 +1,271 @@
+DuelRequests = DuelRequests or {}
+AcceptedDuels = AcceptedDuels or {}
+ActiveDuels = ActiveDuels or {}
+
+RegisterChatCommand({"duel"}, function(player1, arg)
+    local args = string.Explode(" ", arg)
+
+    local kills = args[1]:lower() ~= "accept" and tonumber(args[#args]) or nil
+    local points = args[1]:lower() == "accept" and tonumber(args[#args]) or tonumber(args[#args - 1])
+
+    points = (points ~= nil and math.floor(points)) or nil
+
+    if args[1]:lower() == "accept" and points ~= nil then
+        checkDuelRequest(player1, points)
+        return
+    elseif (args[1]:lower() == "accept" and points == nil) then
+        player1:ChatPrint("[orange]!duel accept [confirm number of points]")
+        return
+    end
+
+    if points == nil or kills == nil then
+        player1:ChatPrint("[orange]!duel [player] [amount of points] [number of kills to win]")
+    else
+        table.remove(args)
+        table.remove(args)
+
+        local player2, count = PlyCount(string.Implode(" ", args))
+        if count == 1 then
+            if (kills <= 20 ) then
+                if (kills >= 1) then
+                    if player1 == player2 then
+                        player1:ChatPrint("[red]You can't duel yourself!")
+                    elseif points >= 1000 then
+                        requestDuel(player1, player2, points, kills)
+                    else
+                        player1:ChatPrint("[red]A duel must be worth a minimum of 1000 points.")
+                    end
+                else
+                    player1:ChatPrint("[red]The win condition can not be less than 1 kill.")
+                end
+            else
+                player1:ChatPrint("[red]The win condition can not be more than 20 kills.")
+            end
+        else
+            player1:ChatPrint("[red]Player " .. string.Implode(" ", args) .. " not found.")
+        end
+    end
+end)
+
+function hasPendingDuel(player1, player2, points)
```

Comment:
This code would be a lot better if the 2 players just got sorted (by ent index maybe)

Reply:
Hmm I _think_ I understand what you mean, lemme see

Reply:
hmm no I don't actually know what you mean by that, I thought you meant like a filter or something but I don't think that would make the code smaller, I do agree its kind of ugly

Reply:
I'm gonna go through the entire code and clean it up, I found a few mistakes already, one thing that I can do better is variable names, also probably use more modular functions.